### PR TITLE
Harsh/mvc restructuring

### PR DESCRIPTION
**Pull request by Harsh528123**
Refactored filters, sorting and database.

### CODE REVIEW

File: app/src/main/java/com/example/softwaresolutionssquad/controllers/FilterController.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    private void updateButtonBackgrounds(boolean isFilterVisible) {
+        int appBlueColor = ContextCompat.getColor(context, R.color.app_blue);
+        ViewCompat.setBackgroundTintList(dateButton, null);
+        ViewCompat.setBackgroundTintList(makeButton, null);
+        ViewCompat.setBackgroundTintList(tagButton, null);
+        ViewCompat.setBackgroundTintList(keywordButton, isFilterVisible ? null : ColorStateList.valueOf(appBlueColor));
+    }
+
+    private void filterResultsByKeywords(String keywordText) {
```
Comment (shanekunj) [inline app/src/main/java/com/example/softwaresolutionssquad/controllers/FilterController.java:100]: should prob remove that if we're not using it

File: app/src/main/java/com/example/softwaresolutionssquad/controllers/DateFilterController.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            @Override
+            public void onClick(View v) {
+                showDatePicker(endDate, startDate, false);
+            }
+        });
+    }
+
+    private void toggleFilters() {
+        // The actual implementation of your hide/show logic
+        // Update this logic according to your requirements
```
Comment (shanekunj) [inline app/src/main/java/com/example/softwaresolutionssquad/controllers/DateFilterController.java:99]: should probably remove these comments

File: app/src/main/java/com/example/softwaresolutionssquad/controllers/InventoryController.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                    fetchedItems.add(item);
+                }
+                listener.onItemsFetched(fetchedItems);
+            } else {
+                listener.onError(task.getException());
+            }
+        });
+    }
+
+    // ... Rest of your InventoryController code...
```
Comment (shanekunj) [inline app/src/main/java/com/example/softwaresolutionssquad/controllers/InventoryController.java:98]: remove

File: app/src/main/java/com/example/softwaresolutionssquad/controllers/MakeFilterController.java [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+            // Reset ListView to default, unfiltered list of items
+            inventoryListView.setAdapter(inventoryListAdapter);
+            // Clear the TextView
+            makesTextView.setText("");
+            // Save all the makes that are present in the list of items
+            ArrayList<String> allMakesList = new ArrayList<>();
+            // Save all the indices of the items selected to filter by
+            ArrayList<Integer> selectedMakesIndices = new ArrayList<>();
+            // Additional logic to reset views and set background colors
```
Comment (shanekunj) [inline app/src/main/java/com/example/softwaresolutionssquad/controllers/MakeFilterController.java:157]: remove line 157

**Review by shanekunj — APPROVED** [review]

**Review by GurtajK — APPROVED** [review]

**Pull request merged**